### PR TITLE
Add PDs for pp-Ref menu

### PR DESCRIPTION
**Pull request by germanfgv**
Add new PDs to handle the PP-reference HLT menu.

Discussion about the details of the PDs can be found here:
[HLT new PDs request cmsTalk Forum](https://cms-talk.web.cern.ch/t/new-datasets-for-upcoming-2023-ppref-hlt-menu/29294)

### CODE REVIEW

File: etc/ReplayOfflineConfiguration.py [hunk 1]
```diff
@@ -609,6 +622,19 @@
                sizePerEvent=2250, # copied from JetHT - should be checked
                scenario=ppScenario)
 
+DATASETS += ["PPRefHardProbes0", "PPRefHardProbes1", "PPRefHardProbes2"]
```
Comment (francescobrivio) [inline etc/ReplayOfflineConfiguration.py:625]: Same comment as above

File: etc/ReplayOfflineConfiguration.py [hunk 2]
```diff
@@ -657,6 +684,23 @@
...
+    addDataset(tier0Config, dataset,
+               do_reco=True,
+               write_reco=False,
+               write_dqm=True,
+               tape_node="T1_US_FNAL_MSS",
+               disk_node="T1_US_FNAL_Disk",
+               alca_producers=["TkAlMuonIsolated", "HcalCalIterativePhiSym", "MuAlCalIsolatedMu",
+                               "HcalCalHO", "HcalCalHBHEMuonProducerFilter",
+                               "SiPixelCalSingleMuonLoose", "SiPixelCalSingleMuonTight",
+                               "TkAlZMuMu", "TkAlDiMuonAndVertex", "TkAlJpsiMuMu", "TkAlUpsilonMuMu"],
```
Comment (francescobrivio) [inline etc/ReplayOfflineConfiguration.py:699]: I suggest to only keep the double muon producers for the DoubleMuon ppRef PDs
```suggestion
               alca_producers=["TkAlZMuMu", "TkAlDiMuonAndVertex", "TkAlJpsiMuMu", "TkAlUpsilonMuMu"],
```

(side note: personally I think it's quite inconvenient to keep swinging from merged Muon to split SingleMuon+DoubleMuon PDs...)

File: etc/ProdOfflineConfiguration.py [hunk 3]
```diff
@@ -729,6 +756,23 @@
...
+    addDataset(tier0Config, dataset,
+               do_reco=True,
+               write_reco=False,
+               write_dqm=True,
+               tape_node="T1_US_FNAL_MSS",
+               disk_node="T1_US_FNAL_Disk",
+               alca_producers=["TkAlMuonIsolated", "HcalCalIterativePhiSym", "MuAlCalIsolatedMu",
+                               "HcalCalHO", "HcalCalHBHEMuonProducerFilter",
+                               "SiPixelCalSingleMuonLoose", "SiPixelCalSingleMuonTight",
+                               "TkAlZMuMu", "TkAlDiMuonAndVertex", "TkAlJpsiMuMu", "TkAlUpsilonMuMu"],
```
Comment (francescobrivio) [inline etc/ProdOfflineConfiguration.py:771]: Same comment as above

**Review by francescobrivio — COMMENTED** [review]
In addition, ss @saumyaphor4252 mentioned in https://cms-talk.web.cern.ch/t/new-datasets-for-upcoming-2023-ppref-hlt-menu/29294/14, the GTs and CMSSW release are not up to date.

File: etc/ProdOfflineConfiguration.py [hunk 4]
```diff
@@ -655,6 +668,19 @@
                sizePerEvent=2250, # copied from JetHT - should be checked
                scenario=ppScenario)
 
+DATASETS += ["PPRefHardProbes0", "PPRefHardProbes1", "PPRefHardProbes2"]
```
[the remark this answers is not in the document]
Comment (germanfgv) [reply etc/ProdOfflineConfiguration.py:671]: Fixing it

File: etc/ReplayOfflineConfiguration.py (hunk 2 above)
Comment (germanfgv) [reply etc/ReplayOfflineConfiguration.py:699]: I guess it is a question for @saumyaphor4252. Which ones should we keep for DoubleMuon PPRef. (I agree about the splitting of Muon PDs)

Comment (germanfgv) [reply etc/ReplayOfflineConfiguration.py:699]: Ok, I confirm we will get riid of hte other producers.

**Comment by germanfgv** [comment]
> In addition, as @saumyaphor4252 mentioned in https://cms-talk.web.cern.ch/t/new-datasets-for-upcoming-2023-ppref-hlt-menu/29294/14, the GTs and CMSSW release are not up to date.

CMSSW and GTs have been already updated in another PR. This merge will not reverse those changes. I rebased in any case.

**Pull request merged**